File: src/client/classes/api/req_res/add-mi-re-kyou-response.ts
'use strict'

import { MiReKyou } from '@/classes/datas/mi-re-kyou'
import { GkillAPIResponse } from '../gkill-api-response'
import { Kyou } from '@/classes/datas/kyou'

export class AddMiReKyouResponse extends GkillAPIResponse {

    added_mirekyou: MiReKyou

    added_kyou: Kyou | null

    constructor() {
        super()
        this.added_mirekyou = new MiReKyou()
        this.added_kyou = null
    }

}

File: src/server/gkill/api/req_res/add_mi_re_kyou_response.go
package req_res

import (
	"github.com/mt3hr/gkill/src/server/gkill/api/message"
	"github.com/mt3hr/gkill/src/server/gkill/dao/reps"
)

type AddMiReKyouResponse struct {
	Messages message.GkillMessages `json:"messages"`

	Errors message.GkillErrors `json:"errors"`

	AddedMiReKyou *reps.MiReKyou `json:"added_mirekyou"`

	AddedKyou *reps.Kyou `json:"added_kyou"`
}
